### PR TITLE
fix: fill the metric type field in the LoadMetaInfo object

### PR DESCRIPTION
**Pull request by SimFG**
- issue: #35960

### CODE REVIEW

File: internal/querycoordv2/task/executor.go [hunk 1]
```diff
@@ -722,3 +727,15 @@ func (ex *Executor) getLoadInfo(ctx context.Context, collectionID, segmentID int
 	loadInfo := utils.PackSegmentLoadInfo(segment, channel.GetSeekPosition(), indexes)
 	return loadInfo, indexInfos, nil
 }
+
+// setMetricTypeForMetaInfo it's a compatibility method for rolling upgrade from 2.3.x to 2.4
+func (ex *Executor) setMetricTypeForMetaInfo(metaInfo *querypb.LoadMetaInfo, indexInfos []*indexpb.IndexInfo) {
```
Comment (czs007) [inline internal/querycoordv2/task/executor.go:732]: no need to keep this

**Comment by xiaofan-luan** [comment]
we don't need to set setMetricTypeForMetaInfo for master?

**Comment by SimFG** [comment]
This error occurs when the querynode delegator of 2.4 sends a request to the querynode worker of 2.3. Considering that the master will be 2.5, which is upgraded from 2.4, to avoid being used by others, no setting is made in the master

**Comment by jaime0815** [comment]
> we don't need to set setMetricTypeForMetaInfo for master?
The only risk is that the search request might fail during the upgrade from 2.3 to 2.5.

**Pull request merged**